### PR TITLE
Added functional tests to test TVM compatibility for Linux images

### PR DESCRIPTION
**Pull request by vinaykotak**
Added test case to test TVM compatibility of gallery images and OS VHDs of distro images that wish to onboard to TVM.

Attaching a test summary indicating that the tests run fine and does not cause regressions.
[TVM-TEST-COMPATIBILITY_summary.log](https://github.com/microsoft/lisa/files/5691676/TVM-TEST-COMPATIBILITY_summary.log)

### CODE REVIEW

**Comment by vinaykotak** [comment]
@LiliDeng @juhlee-microsoft @abhimarathe @harz566 Can you guys please take a look at this PR. I will abandon the previous one as it got a little messed up.

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        wget https://packages.microsoft.com/keys/msopentech.asc
+
+        apt-key add microsoft.asc
+        apt-key add msopentech.asc
+    ;;
+    suse*|opensuse*|sles*|sle_hpc*)
+        zypper ar -t rpm-md -n "packages-microsoft-com-azurecore" --no-gpgcheck https://packages.microsoft.com/yumrepos/azurecore/ azurecore
+    ;;
+    *)
+        LogErr "Distro not supported. Aborting..."
```
Comment (harz566) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: 'Aborted' means something like  a kind of 'failure',  is this 'Skipped' make more sense here? see other SetTestStateXXX  functions.

File: XML/TestCases/FunctionalTests-TVM.xml [hunk 2]
```diff
@@ -14,4 +14,23 @@
...
+    <Tags>tvm</Tags>
+    <Priority>1</Priority>
+    <SetupConfig>
+      <TestLocation>eastus2euap</TestLocation>
+      <SetupType>OneVM</SetupType>
+      <SecureBoot>true</SecureBoot>
+      <vTPM>true</vTPM>
+      <OverrideVMSize>Standard_D2s_v3,=~Standard_D[0-9]+s_v3</OverrideVMSize>
+      <VMGeneration>2</VMGeneration>
+      <ARMImageName>=~0003-com-ubuntu-server-trusted-vm\s18_04-gen2</ARMImageName>
```
Comment (harz566) [inline XML/TestCases/FunctionalTests-TVM.xml:33]: seems the test case content is trying to handle more compatible distro images, like redhat, centos, oracle, debian, ubuntu, suse, etc.  So, this kind of setting for ARMImage constraint doesn't make any sense.

**Comment by harz566** [comment]
Can we add some LISA local test logs of this test case to the description section of this PR?  Just indicate this test case can pass, and has been well tested.

File: XML/TestCases/FunctionalTests-TVM.xml (hunk 2 above)
Comment (vinaykotak) [reply XML/TestCases/FunctionalTests-TVM.xml:33]: Currently there is only one distro and image that supports this test case. None of the other images in gallery support this test. We can add more images as they onboard to TVM. Also, the primary use case of this test is with OS VHDs of the distro images that are requesting to onboard to TVM.

**Comment by vinaykotak** [comment]
> Can we add some LISA local test logs of this test case to the description section of this PR? Just indicate this test case can pass, and has been well tested.

The current image that we are testing this test with does not pass this test. Unfortunately, we do not have the capability to test this test case with any other image or VHD that is compatible with TVM. I can put a test case log in the description that shows that the test runs as expected and does not cause any regressions.

File: Testscripts/Linux/TVM-TestCompatibility.sh (hunk 1 above)
Comment (vinaykotak) [reply Testscripts/Linux/TVM-TestCompatibility.sh]: Done

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+case $DISTRO in
+    redhat*|centos*|oracle*)
+        echo "[packages-microsoft-com-azurecore]" | tee -a /etc/yum.repos.d/azurecore.repo
+        echo "name=packages-microsoft-com-azurecore" | tee -a /etc/yum.repos.d/azurecore.repo
+        echo "baseurl=https://packages.microsoft.com/yumrepos/azurecore/" | tee -a /etc/yum.repos.d/azurecore.repo
+        echo "enabled=1" | tee -a /etc/yum.repos.d/azurecore.repo
+        echo "gpgcheck=0" | tee -a /etc/yum.repos.d/azurecore.repo
+    ;;
+    ubuntu*|debian*)
+        echo "deb [arch=amd64] http://packages.microsoft.com/repos/azurecore/ trusty main" | sudo tee -a /etc/apt/sources.list.d/azure.list
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: release=$(lsb_release -c -s)
echo "deb [arch=amd64] http://packages.microsoft.com/repos/azurecore/ $release main" | sudo tee -a /etc/apt/sources.list.d/azure.list

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        echo "deb [arch=amd64] http://packages.microsoft.com/repos/azurecore/ xenial main" | sudo tee -a /etc/apt/sources.list.d/azure.list
+        echo "deb [arch=amd64] http://packages.microsoft.com/repos/azurecore/ bionic main" | sudo tee -a /etc/apt/sources.list.d/azure.list
+
+        wget https://packages.microsoft.com/keys/microsoft.asc
+        wget https://packages.microsoft.com/keys/msopentech.asc
+
+        apt-key add microsoft.asc
+        apt-key add msopentech.asc
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    ;;
+    suse*|opensuse*|sles*|sle_hpc*)
+        zypper ar -t rpm-md -n "packages-microsoft-com-azurecore" --no-gpgcheck https://packages.microsoft.com/yumrepos/azurecore/ azurecore
+    ;;
+    *)
+        LogErr "Distro not supported. Skipping..."
+        UpdateSummary "Distro not supported. Skipping..."
+        SetTestStateAborted
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: skip case

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+if [ ! -e /usr/bin/mbinfo ]; then
+    LogErr "mbinfo tool is not on the system"
+    SetTestStateAborted
+    exit 0
+fi
+
+
+sudo mbinfo
+
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh:54]: do we need store the output of mbinfo?

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 7]
```diff
@@ -52,7 +49,8 @@ if [ ! -e /usr/bin/mbinfo ]; then
 fi
 
 
-sudo mbinfo
+output=$(sudo mbinfo)
+LogMsg "$output"
 
 if [ $? == 0 ]; then
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: $? is return value for last command

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 8]
```diff
@@ -20,22 +20,19 @@ case $DISTRO in
...
     ;;
     suse*|opensuse*|sles*|sle_hpc*)
         zypper ar -t rpm-md -n "packages-microsoft-com-azurecore" --no-gpgcheck https://packages.microsoft.com/yumrepos/azurecore/ azurecore
     ;;
     *)
-        LogErr "Distro not supported. Skipping..."
-        UpdateSummary "Distro not supported. Skipping..."
+        LogErr "Distro not supported. Skipping test case..."
+        UpdateSummary "Distro not supported. Skipping test case..."
         SetTestStateAborted
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: SetTestStateAborted => SetTestStateSkipped

File: Testscripts/Linux/TVM-TestCompatibility.sh (hunk 7 above)
Comment (LiliDeng) [reply Testscripts/Linux/TVM-TestCompatibility.sh]: output=$(sudo mbinfo)
ret=$?
LogMsg "$output"
if [ $ret == 0 ]; then

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        echo "enabled=1" | tee -a /etc/yum.repos.d/azurecore.repo
+        echo "gpgcheck=0" | tee -a /etc/yum.repos.d/azurecore.repo
+    ;;
+    ubuntu*|debian*)
+        release=$(lsb_release -c -s)
+        echo "deb [arch=amd64] http://packages.microsoft.com/repos/azurecore/ $release main" | sudo tee -a /etc/apt/sources.list.d/azure.list
+
+        wget -qO https://packages.microsoft.com/keys/microsoft.asc | apt-key add microsoft.asc
+        wget -qO https://packages.microsoft.com/keys/msopentech.asc | apt-key add msopentech.asc
+
```
Comment (LiliDeng) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: wget -qO - https://packages.microsoft.com/keys/microsoft.asc | apt-key add -
 wget -qO - https://packages.microsoft.com/keys/msopentech.asc | apt-key add -

Comment (LiliDeng) [reply Testscripts/Linux/TVM-TestCompatibility.sh]: Could you please run the commands?
```
wget -qO - https://packages.microsoft.com/keys/microsoft.asc | apt-key add microsoft.asc
gpg: can't open 'microsoft.asc': No such file or directory
```

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**